back/app/services/local_project_copy_service.rb

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+# Copies a project within a tenant.
+class LocalProjectCopyService
+  def copy(source_project)
+    new_title_multiloc = add_suffix_to_title(source_project.title_multiloc)
+
+    options = {
+      local_copy: true,
+      include_ideas: false,
+      anonymize_users: false,
+      new_title_multiloc: new_title_multiloc,
+      timeline_start_at: Time.now.to_s,
+      new_publication_status: 'draft'
+    }
+
+    template = AdminApi::ProjectCopyService.new.export source_project, **options
+    folder_id = ProjectFolders::Folder.find(source_project.folder_id) if source_project.folder_id
+    copied_project = AdminApi::ProjectCopyService.new.import(template, folder: folder_id, local_copy: true)
+
+    copy_project_visibility_permission_groups(source_project, copied_project)
+
+    # Copy actions groups_permissions of project or project phases. For example, groups that can 'Vote on ideas'.
+    copy_project_and_phases_actions_groups_permissions(source_project, copied_project)
+
+    source_project.allowed_input_topics.each do |topic|
+      input_topic = ProjectsAllowedInputTopic.find_by(topic_id: topic.id)
```

Comment:
Do we need to query a `projects_allowed_input_topic` by `project_id` too?

Reply:
I think this pattern can be used for all entities

```ruby
source_project.allowed_input_topics.each { _.dup.update!(project_id: copied_project.id) }
source_project.areas_projects.each { _.dup.update!(project_id: copied_project.id) }
source_project.projects_topics.each { _.dup.update!(project_id: copied_project.id) }
```

Reply:
Nice pattern! Thanks.